cinema/views.py

```diff
@@ -56,3 +95,42 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        movie = self.request.query_params.get("movie")
+        date = self.request.query_params.get("date")
+
+        if self.action in ("list", "retrieve"):
+            self.queryset = self.queryset.select_related("movie",
+                                                         "cinema_hall")
```

Comment:
although acceptable by PEP8, this is rarely used style - the more widely used is newline after opening parenthesis